lib/go-qmstr/cli-qmstr/run.go

```diff
@@ -0,0 +1,195 @@
+package cliqmstr
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	keepTmpDirectories bool                                                  // Keep intermediate files
+	instdir            string                                                // create instrumentation in this dir
+	wrappedCmds        = []string{"gcc", "g++", "ar", "ld", "as", "objcopy"} // constant (which Go does not do with arrays)
+)
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Run a shell command with QMSTR instrumentation",
+	Long: `run executes the argument after applying temporary QMSTR instrumentation.
+For example, "qmstr make" sets up a QMSTR instrumented environment, then executes  "make", and finally restores the environment and deletes all temporary files.
+The environment variable QMSTR_INSTRUMENTATION_HOME will be defined to point to the QMSTR instrumentation directory.`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 && !keepTmpDirectories {
+			return fmt.Errorf("no command specified")
+		}
+		return nil
+	},
+	Run: executeRun,
+}
+
+func init() {
+	rootCmd.AddCommand(runCmd)
+	runCmd.Flags().SetInterspersed(false) // this stops flag parsing at the first argument, allowing for e.g. "ls -la" as payload
```

Comment:
What if we ever need to add a second argument?
What was wrong with proper gnu-style as we used it before and stop parsing with --? 
see https://github.com/QMSTR/qmstr-demo/blob/7383eb658c63f78c33c17d8a15db29e9eb172004/demos/curl/build.sh#L31
https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

Reply:
Discussed and resolved.